### ISSUE TITLE
pom.xml <sourceDirectory> not respected?

### ISSUE THREAD

**Issue by nathanaeljones** [teamed/qulice#382]
We specify <sourceDirectory> in our pom.xml. https://github.com/imazen/sharpen/blob/develop/pom.xml#L55

Only FindBugs sees any files; we get the following output:

     [INFO] No files to check with Checkstyle
     [INFO] No files to check with PMD
     [INFO] No PMD violations found in 0 files
     [INFO] CodeNarc not required since no sources in /Users/nathanael/Documents/sharpen/src

**Comment by yegor256** [comment]
@nathanaeljones that's true, it's a bug in our Maven plugin, we are looking into `src` only: https://github.com/teamed/qulice/blob/master/qulice-maven-plugin/src/main/java/com/qulice/maven/DefaultMavenEnvironment.java#L211

thanks for reporting, we'll try to fix this soon!

**Comment by davvd** [comment]
I'll ask someone to take care of this task soon

**Comment by davvd** [comment]
@cyberone it's yours now, please proceed keeping in mind [our principles](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html). Feel free to ask any technical questions right here in the ticket. This task's budget is **30 mins**. This is exactly how much will be paid when the problem explained above is solved. See [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for more information

**Comment by cyberone** [comment]
@yegor256 can you please help? I have no idea why build fails in modules I didn't modify.

**Comment by yegor256** [comment]
@cyberone it fails in master branch? report a bug.

**Comment by cyberone** [comment]
@davvd please find somebody else for this task

**Comment by davvd** [comment]
> @davvd please find somebody else for this task

@cyberone I will try to find someone else

**Comment by krzyk** [comment]
@davvd this is postponed

**Comment by davvd** [comment]
> @davvd this is postponed

@krzyk thanks, I added "postponed" label

**Comment by davvd** [comment]
> @davvd this is postponed

@krzyk all right, we'll find someone else for this task

**Comment by yegor256** [comment]
Fixed in 44c2d2ebd.

Root cause: `DefaultMavenEnvironment.files()` hardcoded the `src` directory under basedir, so any `<sourceDirectory>`, `<testSourceDirectory>`, or `<resources>` declared in `pom.xml` was ignored. Projects that placed their sources outside `src/` (like the one originally reported) ended up with "No files to check" messages from Checkstyle, PMD, and CodeNarc.

The method now pulls directories from `project.getCompileSourceRoots()`, `project.getTestCompileSourceRoots()`, and `build.getResources()/getTestResources()`, resolving relative paths against the project basedir. It falls back to `basedir/src` only when the project exposes no directories at all, which preserves the old behavior for minimal test stubs.

Three new tests in `DefaultMavenEnvironmentTest` cover custom `<sourceDirectory>`, custom `<testSourceDirectory>`, and custom resources.
